H/HDF5/build_tarballs.jl

```diff
@@ -94,6 +94,7 @@ platforms = [
 # The products that we will ensure are always built
 products = [
     LibraryProduct("libhdf5", :libhdf5),
+    LibraryProduct("libhdf5_hl", :libhdf5_hl),
```

Comment:
This is very bad:
```suggestion
    LibraryProduct(["libhdf5_hl", "libhdf5_hl-3aef14e1", "libhdf5_hl-db841637"], :libhdf5_hl),
```
and I'm not even sure it'll work

Reply:
What you did is clearly better, thanks!